ludwig/encoders/image_encoders.py

```diff
@@ -36,9 +36,40 @@
 logger = logging.getLogger(__name__)
 
 
+@DeveloperAPI
+class ImageEncoder(Encoder):
+    @classmethod
+    def requires_equal_dimensions(cls) -> bool:
```

Comment:
This is great! To reduce boilerplate for non-ViT, you could have these methods default to `False`, `None`, and `None`. That way you only need to override for the ones that require it.